source/MaterialXFormat/XmlIo.cpp

```diff
@@ -268,6 +282,7 @@ void readFromXmlFile(DocumentPtr doc, const string& filename, const string& sear
     {
         doc->setSourceUri(filename);
     }
+    documentFromXml(doc, xmlDoc, searchPath, readOptions);
```

Comment:
This must be done after the source URI is set since the function uses that to find the path.